pkg/3scale/amp/operator/upgrade.go

```diff
@@ -394,6 +405,124 @@ func (u *UpgradeApiManager) findDeploymentTriggerOnImageChange(triggerPolicies [
 	return result, nil
 }
 
+func (u *UpgradeApiManager) upgradeBackendRouteEnv() (reconcile.Result, error) {
+	system, err := System(u.apiManager, u.Client())
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	res, err := u.upgradeSystemAppBackendRouteEnv(system.AppDeploymentConfig())
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	res, err = u.upgradeSidekiqBackendRouteEnv(system.SidekiqDeploymentConfig())
+	if res.Requeue || err != nil {
+		return res, err
+	}
+
+	return reconcile.Result{}, nil
+}
+
+func (u *UpgradeApiManager) upgradeSystemAppBackendRouteEnv(desired *appsv1.DeploymentConfig) (reconcile.Result, error) {
+	existing := &appsv1.DeploymentConfig{}
+	err := u.Client().Get(context.TODO(), types.NamespacedName{Name: desired.Name, Namespace: u.apiManager.Namespace}, existing)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if len(existing.Spec.Template.Spec.Containers) != 3 {
+		return reconcile.Result{}, fmt.Errorf("DeploymentConfig %s spec.template.spec.containers length is %d, should be 3",
+			existing.Name, len(existing.Spec.Template.Spec.Containers))
+	}
+
+	desiredName := common.ObjectInfo(desired)
+
+	update := false
+	for idx := 0; idx < 3; idx++ {
+		existingContainer := &existing.Spec.Template.Spec.Containers[idx]
+		desiredContainer := &desired.Spec.Template.Spec.Containers[idx]
+		desiredBackendRouteEnvVar, ok := helper.FindEnvVar(desiredContainer.Env, "BACKEND_ROUTE")
+		if !ok {
+			panic(fmt.Sprintf("%s desired spec.template.spec.containers env var '%s' does not exist", desiredName, "BACKEND_ROUTE"))
```

Comment:
I knew this shouldn't happen but maybe returning an error so we don't completely crash the operator?